docs/includes/security/providers/oidc.adoc

```diff
@@ -79,6 +79,63 @@ introspection endpoint depending on configuration
 5. We validate that we have sufficient scopes to proceed, and return `403` if not
 6. Handling is returned to security to process other security providers
 
+=== Multiple tenants
+OIDC provider also supports multiple tenants. To enable this feature it is required to do several steps.
+
+1. To enable default multi tenant support, add `multi-tenant: true` option to your
+OIDC provider configuration
+2. Specify desired way of how to provide tenant name information. This is done over adding
+`tenant-id-style` config option. More information can be found in the table below
+3. Add `tenants` section to the OIDC provider config
+
+[source, yaml]
+----
+tenants:
+   - name: "example-tenant"
+     ... tenant configuration options
+----
+
+There are 4 ways how to provide required tenant information to Helidon by default.
+
+.Possible `tenant-id-style` configuration options
+[cols="2,4a,3a"]
+|===
+|key |description |additional config options
+
+|`host-header` |Tenant configuration will be selected based on your host present in the Host header value |{nbsp}
+
+|`domain`
+|Similar to the `host-header` style, but now the tenant name is identified just as
+a part of the host name. By default, it selects the third domain level.
+
+Example: Host header value from inbound request is `my.helidon.com` -> domain level 3 is `my`,
+domain level 2 is `helidon` and domain level 1 is `com`
+|[source, yaml]
+----
+tenant-id-domain-level: <domain level>
+----
+
+|`token-handler`
+|Tenant name information is expected to be provided via configured custom header value
+|[source, yaml]
+----
+tenant-id-handler:
+  header: "my-custom-header"
+----
+
+|`none`|No tenant name finding is used. Default tenant name `@default` is used instead|
+|===
+
+It is also possible to implement custom way of discovering the tenant name and tenant configuration.
```

Comment:
You can also implement a custom way of discovering the tenant name and tenant configuration. The custom tenant name discovery from request can be done by implementing SPI: